### PR TITLE
Rounding law support

### PR DESCRIPTION
**Pull request by celiot-IDM**
This PR introduces support for Rounding Laws. Rounding Laws are the rules pacehrh uses to determine if and when computed values are rounded. The Rounding Law is set with the SetRoundingLaw() function. There are three rounding laws - "early", "late", and "none" - and the differences between them are described in the documentation to SetRoundingLaw().

The first applications of RoundingLaw = "none" (which preserves full decimal precision) are to population predictions, allowing more precise unit testing of population functions. 

This work is in support of issues #67 and #133, but doesn't close them.

### CODE REVIEW

File: pacehrh/tests/testthat/test-001-pace_config.R [hunk 1]
```diff
@@ -199,13 +199,16 @@ test_that("Global configuration: check-and-load", {
 test_that("Global configuration: SetGlobalStartEndYears", {
   e <- pacehrh:::GPE
 
-  withr::defer(e$startYear <- originalStartYear)
+  local_vars("startYear", envir = e)
+  local_vars("endYear", envir = e)
+  local_vars("endYear", envir = e)
```
Comment (rhanIDM) [inline pacehrh/tests/testthat/test-001-pace_config.R:204]: Are line 203 and 204 duplicates?

**Comment by celiot-IDM** [comment]
Good catch! I’ll repair shortly.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: pacehrh/R/pace_config.R [hunk 2]
```diff
@@ -167,3 +167,38 @@ SetGlobalStartEndYears <- function(start = 2020, end = 2040, shoulderYears = 1)
...
   return(invisible(NULL))
 }
+
+#' Set Rounding Law
+#'
+#' The rounding law determines how the system handles rounding for
+#' processes that in nature would always return integer counts, such as
+#' births.
+#'
+#' For example: Take 35 age groups, each containing 100 fertile women, and
```
Comment (MeWu-IDM) [inline pacehrh/R/pace_config.R:177]: is the early / late concept only applies to the fertility rate? For other calculation it seems they don't matter (only none option matters) I feel that this description should clarify that also mentioned that the default is "early" if using NULL

**Review by MeWu-IDM — APPROVED** [review]
tested rounding option, it looks good but I think the description should cover other calculations which use roundingLaw without "early/late" condition

**Pull request merged**